### PR TITLE
Canny source doc

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Added doc for Canny source

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/cloud-apps/canny.mdx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+6. Finally, click on **Add**. 
+
+## Event transformation
+
+RudderStack ingests the events from Canny after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:
+
+| Canny property  | RudderStack property| Notes |
+| :------------------ | :------------- |:------------- |
+| `created`  | `timestamp`,`originalTimestamp`    | - |
+| `object`        | `properties.objectType`        | -|
```
Comment (rohithkaza) [inline docs/sources/cloud-apps/canny.mdx:37]:
```suggestion
| `objectType`        | `properties.objectType`        | -|
```

Comment (rohithkaza) [reply docs/sources/cloud-apps/canny.mdx:37]: object-> properties mapping is missing

File: docs/sources/cloud-apps/canny.mdx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+| `created`  | `timestamp`,`originalTimestamp`    | - |
+| `object`        | `properties`        | -|
+| `objectType`        | `properties.objectType`        | -|
+| `type`    | `event`  | -|
+| `object.author.userId`     | `userId`   |- |
+| sha256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
+| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
+| `object.voter.userId`    | `userId`  | Only if `type`is `vote.created`/`vote.deleted` |
+| `object.voter.id`        | `context.externalId[0].value` | Only if `type`is `vote.created`/`vote.deleted`|
+| `cannyId`    | `context.externalId[0].type`  |- |
```
Comment (ItsSudip) [inline docs/sources/cloud-apps/canny.mdx:45]: 'cannyId' does not belong to Canny, we are assigning the type of externalId as `cannyUserId`.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by rohithkaza — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

File: docs/sources/cloud-apps/canny.mdx [hunk 3]
```diff
@@ -0,0 +1,184 @@
+---
+title: "Canny"
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:2]:
```suggestion
title: "Canny Source"
```

Comment (ameypv-rudder) [reply docs/sources/cloud-apps/canny.mdx:2]: To help with the search results.

File: docs/sources/cloud-apps/canny.mdx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+1. Go to your [RudderStack dashboard](https://app.rudderstack.com/) and click on **Add Source**. Then, select **Canny** from the list of **Event Stream** sources, as shown:
+
+<img src="../../assets/event-stream-sources/canny-1.png" alt="Canny source" />
+
+2. Assign a name to your source and click on **Next**.
+3. The Canny source is now configured. Note the **Webhook URL** in the **Settings** tab.
+
+<img src="../../assets/event-stream-sources/canny-2.png" alt="Canny source webhook URL" />
+
+4. Go to your Canny account and navigate to **Settings** > **API & Webhooks** under your profile.
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:23]:
```suggestion
4. Go to your <a href="https://canny.io/login">Canny account</a> and navigate to **Settings** > **API & Webhooks** under your profile.
```

File: docs/sources/cloud-apps/canny.mdx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+5. Enter the **Webhook URL** under **Webhooks** section, as shown:
+
+<img src="../../assets/event-stream-sources/webhook-url-canny.png" alt="Canny source webhook URL" />
+
+6. Finally, click on **Add**. 
+
+## Event transformation
+
+RudderStack ingests the events from Canny after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:32]:
```suggestion
RudderStack ingests the Canny events after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:
```

File: docs/sources/cloud-apps/canny.mdx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+| Canny property  | RudderStack property| Notes |
+| :------------------ | :------------- |:------------- |
+| `created`  | `timestamp`,`originalTimestamp`    | - |
+| `object`        | `properties`        | excluding `author`/`voter` from properties|
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:37]:
```suggestion
| `object`        | `properties`        | RudderStack excludes `author`/`voter` from the properties. |
```

Comment (ameypv-rudder) [reply docs/sources/cloud-apps/canny.mdx:37]: Please verify if this is accurate.

File: docs/sources/cloud-apps/canny.mdx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+| `objectType`        | `properties.objectType`        | -|
+| `type`    | `event`  | -|
+| `object.author`    | `context.traits`  | excluding `userId` and `id` |
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:40]:
```suggestion
| `object.author`    | `context.traits`  | RudderStack excludes `userId` and `id`. |
```

File: docs/sources/cloud-apps/canny.mdx [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+| `object.author.userId`     | `userId`   |- |
+| sha256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
+| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:43]:
```suggestion
| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present **and** `type`is `vote.created`/`vote.deleted`|
```

File: docs/sources/cloud-apps/canny.mdx [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+| Canny property  | RudderStack property| Notes |
+| :------------------ | :------------- |:------------- |
+| `created`  | `timestamp`,`originalTimestamp`    | - |
+| `object`        | `properties`        | excluding `author`/`voter` from properties|
+| `objectType`        | `properties.objectType`        | -|
+| `type`    | `event`  | -|
+| `object.author`    | `context.traits`  | excluding `userId` and `id` |
+| `object.author.userId`     | `userId`   |- |
+| sha256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:42]:
```suggestion
| SHA-256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
```

File: docs/sources/cloud-apps/canny.mdx (hunk 8 above)
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:43]:
```suggestion
| SHA-256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
```

File: docs/sources/cloud-apps/canny.mdx (hunk 9 above)
Comment (ameypv-rudder) [reply docs/sources/cloud-apps/canny.mdx:42]: Are the brackets required here? Just confirming.

File: docs/sources/cloud-apps/canny.mdx [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
+| `object.voter.userId`    | `userId`  | Only if `type`is `vote.created`/`vote.deleted` |
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:44]:
```suggestion
| `object.voter.userId`    | `userId`  | Only if `type` is `vote.created`/`vote.deleted` |
```

File: docs/sources/cloud-apps/canny.mdx [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+| `object.voter.id`        | `context.externalId[0].value` | Only if `type`is `vote.created`/`vote.deleted`|
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:45]:
```suggestion
| `object.voter.id`        | `context.externalId[0].value` | Only if `type` is `vote.created`/`vote.deleted`|
```

File: docs/sources/cloud-apps/canny.mdx [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+| `cannyUserId`    | `context.externalId[0].type`  |- |
+| `object.author.id`     | `context.externalId[0].id`   |- |
+
+
+Canny supports the following events:
```
Comment (ameypv-rudder) [inline docs/sources/cloud-apps/canny.mdx:50]:
```suggestion
Canny supports ingesting the following events:
```

**Review by ameypv-rudder — APPROVED** [review]

File: docs/sources/cloud-apps/canny.mdx (hunk 3 above)
Comment (ishachopra0401) [reply docs/sources/cloud-apps/canny.mdx:2]: Should we add it in alias tag rather? search results will be catered, and consistency across all source docs will be maintained too. WDYT

Comment (ameypv-rudder) [reply docs/sources/cloud-apps/canny.mdx:2]: Makes sense!

**Pull request merged**